blockdata/constraints.lisp

```diff
@@ -1,147 +1,380 @@
+
+(defun (euc-call w a b)
+  (begin (eq! (shift EUC_FLAG w) 1)
+         (eq! (shift ARG_1_HI w) a)
+         (eq! (shift ARG_1_LO w) a)))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;                          ;;
+;;  2.1 Shorthands          ;;
+;;                          ;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defun    (flag-sum)    (+ IS_CB
+                           IS_TS
+                           IS_NB
+                           IS_DF
+                           IS_GL
+                           IS_ID
+                           IS_BF))
+
+(defun    (wght-sum)    (+ ( * 1 IS_CB)
+                           ( * 2 IS_TS)
+                           ( * 3 IS_NB)
+                           ( * 4 IS_DF)
+                           ( * 5 IS_GL)
+                           ( * 6 IS_ID)
+                           ( * 7 IS_BF)))
+
+(defun    (inst-sum)    (+ (* EVM_INST_COINBASE   IS_CB)
+                           (* EVM_INST_TIMESTAMP  IS_TS)
+                           (* EVM_INST_NUMBER     IS_NB)
+                           (* EVM_INST_DIFFICULTY IS_DF)
+                           (* EVM_INST_GASLIMIT   IS_GL)
+                           (* EVM_INST_CHAINID    IS_ID)
+                           (* EVM_INST_BASEFEE    IS_BF)))
+
+(defun    (ct-max-sum)    (+ (* (- CT_MAX_CB 1) IS_CB)
+                             (* (- CT_MAX_TS 1) IS_TS)
+                             (* (- CT_MAX_NB 1) IS_NB)
+                             (* (- CT_MAX_DF 1) IS_DF)
+                             (* (- CT_MAX_GL 1) IS_GL)
+                             (* (- CT_MAX_ID 1) IS_ID)
+                             (* (- CT_MAX_BF 1) IS_BF)))
+
+(defun    (phase-entry)    (+ (* (- 1 IS_CB) (shift IS_CB 1))
+                              (* (- 1 IS_TS) (shift IS_TS 1))
+                              (* (- 1 IS_NB) (shift IS_NB 1))
+                              (* (- 1 IS_DF) (shift IS_DF 1))
+                              (* (- 1 IS_GL) (shift IS_GL 1))
+                              (* (- 1 IS_ID) (shift IS_ID 1))
+                              (* (- 1 IS_BF) (shift IS_BF 1))))
+
+(defun    (same-phase)     (+ (* IS_CB (shift IS_CB 1))
+                              (* IS_TS (shift IS_TS 1))
+                              (* IS_NB (shift IS_NB 1))
+                              (* IS_DF (shift IS_DF 1))
+                              (* IS_GL (shift IS_GL 1))
+                              (* IS_ID (shift IS_ID 1))
+                              (* IS_BF (shift IS_BF 1))))
+
+(defun    (allowable-transitions)     (+ (* IS_CB (shift IS_TS 1))
+                                         (* IS_TS (shift IS_NB 1))
+                                         (* IS_NB (shift IS_DF 1))
+                                         (* IS_DF (shift IS_GL 1))
+                                         (* IS_GL (shift IS_ID 1))
+                                         (* IS_ID (shift IS_BF 1))
+                                         (* IS_BF (shift IS_CB 1))))
```

Comment:
I leave it to your discretion @amkCha and @lorenzogentile404 , but you can use
```lisp
(next XXX)
```
rather than
```lisp
(shift XXX 1)
```